pylint/checkers/unicode.py

```diff
@@ -0,0 +1,701 @@
+                "invalid-unicode-codec",
+                # Currently Nodes will lead to crashes of pylint
+                # node=node,
+                line=invalid_codec_line,
+                end_lineno=invalid_codec_line,
+                confidence=interfaces.HIGH,
+                col_offset=None,
+                end_col_offset=None,
+            )
+
+        # Default to UTF-8 if no codec was found
+        codec = codec or "UTF-8"
+
+        # Check for Bidirectional Unicode
+        with node.stream() as stream:
+            for (lineno, line) in enumerate(_fix_utf16_32_line_stream(stream, codec)):
+
+                # Look for Bidirectional Unicode
+                for dangerous in self._BIDI_UNICODE:
+                    if _cached_encode_search(dangerous, codec) in line:
+                        # Note that we don't add a col_offset on purpose:
+                        #   Using these unicode characters it depends on the editor
+                        #   how it displays the location of characters in the line.
+                        #   So we mark the complete line.
+                        self.add_message(
+                            "bidirectional-unicode",
+                            # Currently Nodes will lead to crashes of pylint
+                            # node=node,
+                            line=lineno + 1,
+                            end_lineno=lineno + 1,
+                            # We mark the complete line, as bidi controls make it hard
+                            # to determine the correct cursor position within an editor
+                            col_offset=0,
+                            end_col_offset=line_length(line, codec),
+                            confidence=interfaces.HIGH,
+                        )
+                        # We look for bidirectional unicode only once per line
+                        # as we mark the complete line anyway
+                        break
+
+                if lineno == 0:
+                    line = _remove_bom(line, codec)
+
+                # Look for bad ASCII chars
+                matches = self._find_line_matches(line, codec)
+                for col, char in matches.items():
+                    self.add_message(
+                        char.human_code(),
+                        # Currently Nodes will lead to crashes of pylint
+                        # node=node,
+                        line=lineno + 1,
+                        end_lineno=lineno + 1,
+                        confidence=interfaces.HIGH,
+                        col_offset=col,
+                        end_col_offset=col + len(char.unescaped),
+                    )
+
+
+class _AsciiOnlyCheckedNode(Protocol):
+    _is_ascii_only: bool
+
+
+class NoneAsciiNamesChecker(BaseChecker):
+    """A very strict version of the name checker that only allows ASCII for everything!
+
+    If your programming guideline defines that your are programming in English,
```

Comment:
If had not seen this previously. I think this should definitely move to `extensions`. We get issue reports with screenshots in Asian languages and code examples somewhat regularly. We can't assume that all of our users are programming in English, so this should probably be turned of by default.

Reply:
This would be change of the current behaviour. It was a part of the BaseChecker and I just reimplemented it to be more strict on what to find.

Reply:
I changed the docstring and the message, to make it clearer.

The requirement for ASCII chars for **names** is not only done in this function but also within the NameChecker.
The Name checker doesn't allow foreign language names by default as well.

Reply:
Hmm I didn't know that. Guess that happens which these larger projects.

Well, no reason to change something that has been the default for years now. We would probably make another decision now, but this is fine! Thanks for your explanation!